File: android/src/main/java/expo/modules/appleintelligence/AppleIntelligenceModule.kt
package expo.modules.appleintelligence

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition
import expo.modules.kotlin.Promise

/**
 * Android stub for Apple Intelligence module.
 * Apple Intelligence is only available on iOS devices, so this module
 * returns unavailable status on Android.
 */
class AppleIntelligenceModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("AppleIntelligence")

    AsyncFunction("isAvailable") { 
      false
    }

    AsyncFunction("getAvailabilityStatus") {
      mapOf(
        "status" to "unsupported",
        "isAvailable" to false,
        "message" to "Apple Intelligence is only available on iOS devices"
      )
    }

    AsyncFunction("generateResponse") { _: String ->
      mapOf(
        "success" to false,
        "error" to "Apple Intelligence is only available on iOS devices"
      )
    }

    AsyncFunction("createSession") { _: String? ->
      throw Exception("Apple Intelligence is only available on iOS devices")
    }

    AsyncFunction("sendSessionMessage") { _: String, _: String ->
      mapOf(
        "success" to false,
        "error" to "Apple Intelligence is only available on iOS devices"
      )
    }

    AsyncFunction("resetSession") { _: String ->
      // No-op on Android
    }
  }
}

File: src/AppleIntelligenceModule.ts
import { NativeModule, requireNativeModule } from 'expo-modules-core';

import type { AvailabilityStatus, GenerateResponseResult } from './AppleIntelligence';

declare class AppleIntelligenceModuleType extends NativeModule {
  isAvailable(): Promise<boolean>;
  getAvailabilityStatus(): Promise<AvailabilityStatus>;
  generateResponse(prompt: string): Promise<GenerateResponseResult>;
  createSession(systemPrompt?: string): Promise<string>;
  sendSessionMessage(sessionId: string, message: string): Promise<GenerateResponseResult>;
  resetSession(sessionId: string): Promise<void>;
}

// This call throws if the native module is not available
// We handle this gracefully in AppleIntelligence.ts
export const AppleIntelligenceModule =
  requireNativeModule<AppleIntelligenceModuleType>('AppleIntelligence');

export default AppleIntelligenceModule;

